### PR TITLE
Issue #3003 Use filesystem notification to update systray menu

### LINKED ISSUE #3003: Profile stays shown in system tray after it is deleted
### General information

  * Minishift version: minishift v1.27.0+707887e
  * OS: Windows 7
  * Hypervisor: VirtualBox

### Steps to reproduce

  1. minishift profile new coolprofile
  2. minishift start
  3. see it is shown in the systray
  4. minishift profile delete coolprofile
  5. profile stays listed in systray

### Logs

You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #3003 

Steps to test the pull request

  1. Download artifacts for [darwin](https://5814-62728051-gh.circle-artifacts.com/0/ms/darwin-amd64/systemtray/minishift) or [windows](https://5814-62728051-gh.circle-artifacts.com/0/ms/windows-amd64/systemtray/minishift.exe)
  2. `minishift start`
  3. `minishift profile set test` # The new profile should be shown in tray instantly
  4. `minishift profile delete test` # test profile should disappear from tray instantly

### CODE REVIEW

File: cmd/minishift/state/instance_dirs.go [hunk 1]
```diff
@@ -53,6 +54,7 @@ func GetMinishiftDirsStructure(baseDir string) *MinishiftDirs {
 		Logs:         filepath.Join(baseDir, "logs"),
 		Tmp:          filepath.Join(baseDir, "tmp"),
 		Config:       filepath.Join(baseDir, "config"),
+		Profiles:     filepath.Join(baseDir, "profiles"),
```
Comment (anjannath) [inline cmd/minishift/state/instance_dirs.go]: Did we not include the profiles directory here on purpose? @minishift/minishift-dev

Comment (LalatenduMohanty) [reply cmd/minishift/state/instance_dirs.go]: @anjannath Ah it should have the profiles directory, Checking the code.

Comment (anjannath) [reply cmd/minishift/state/instance_dirs.go]: @LalatenduMohanty I need this directory to be present even if no profiles have yet been created, as i put a inotify watcher on the directory to monitor it for creation and deletion of profiles.

Comment (LalatenduMohanty) [reply cmd/minishift/state/instance_dirs.go]: @anjannath We can not add `profiles` to the structure as no profiles except the default profile need to create the directory.  Also if we accept this change every profile will have an extra `profiles` directory e.g. `~/.minishift/profiles/<profile name>/profiles` which will be never used. I would suggest to add code in root.go to create ~/.minishift/profiles directory.

Comment (LalatenduMohanty) [reply cmd/minishift/state/instance_dirs.go]: @anjannath We can not add profiles to the structure as no profiles except the default profile need to create the directory. Also if we accept this change every profile will have an extra profiles directory e.g. ~/.minishift/profiles/<profile name>/profiles which will be never used. I would suggest to add code in root.go to create ~/.minishift/profiles directory.

File: pkg/minishift/systemtray/systemtray.go [hunk 2]
```diff
@@ -118,57 +118,53 @@ func getStatus(profileName string) int {
...
-				go startStopHandler(icon.Stopped, profile, ma.stop, STOP_PROFILE)
-			}
-		}
+// Add newly created profiles and remove deleted profiles from tray
+func updateTrayMenu() {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		fmt.Println("Failed to create watcher:", err)
+	}
+	profilesBaseDir := filepath.Join(constants.GetMinishiftHomeDir(), "profiles")
```
Comment (LalatenduMohanty) [inline pkg/minishift/systemtray/systemtray.go]: Reconstructing the profile directory path here is not ideal IMO but there are places we have done this. Also this code is going for maintenance so I am kind of ok with it.

Comment (anjannath) [reply pkg/minishift/systemtray/systemtray.go]: @LalatenduMohanty I can add it to the constants and also change it in other places wherever needed, if it is ok to do it in this PR?

Comment (LalatenduMohanty) [reply pkg/minishift/systemtray/systemtray.go]: @anjannath yes, that should be ok. But ideally it should not be a constant but a function which can the environment variable for Minishift home in to account.

**Comment by anjannath** [comment]
@LalatenduMohanty All the changes have been made, PTAL

**Pull request merged**